profilingdemo/source/java/src/org/kaaproject/kaa/demo/profiling/ProfilingDemo.java

```diff
@@ -0,0 +1,198 @@
+/**
+ *  Copyright 2014-2016 CyberVision, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.kaaproject.kaa.demo.profiling;
+
+import org.kaaproject.examples.pager.PagerClientProfile;
+import org.kaaproject.examples.pager.PagerConfiguration;
+import org.kaaproject.kaa.client.DesktopKaaPlatformContext;
+import org.kaaproject.kaa.client.Kaa;
+import org.kaaproject.kaa.client.KaaClient;
+import org.kaaproject.kaa.client.KaaClientProperties;
+import org.kaaproject.kaa.client.SimpleKaaClientStateListener;
+import org.kaaproject.kaa.client.configuration.base.ConfigurationListener;
+import org.kaaproject.kaa.client.configuration.base.SimpleConfigurationStorage;
+import org.kaaproject.kaa.client.profile.ProfileContainer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * A demo application class that use the Kaa endpoint profiling and grouping API.
+ *
+ * @author Maksym Liashenko
+ */
+public class ProfilingDemo {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ProfilingDemo.class);
+
+    public static final int KAA_CLIENT_NUMBER = 3;
+    /**
+     * In this dir all Kaa clients creates it's keys for starting
+     */
+    public static final String PROPERTIES_OUT_KEYS_DIR = "./res/out/kaaTempDir";
+
+    private static ArrayList<KaaClient> kaaClients = new ArrayList<>();
+    private static ArrayList<InnerData> profiles = new ArrayList<>();
+
+    public static void main(String[] args) throws IOException {
+        LOG.info("Profiling demo started");
+
+        for (int i = 0; i < KAA_CLIENT_NUMBER; i++) {
+            try {
+                startKaaClient(i);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        LOG.info("--= Press any key to exit =--");
```

Comment:
Wait for all `kaaClient`s to start before printing this line. For example use `TimeUnit.SECONDS,sleep(5)`.